eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,71 @@
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+function InstallRetrieveCodeOwnersTool() {
+  # Check if the retrieve-codeowners tool exsits or not.
+  if (Get-Command "$ToolPath/$ToolCommandName" -errorAction SilentlyContinue) {
+    return "$ToolPath/$ToolCommandName"
   }
-}
+  if (!$ToolPath -or !(Test-Path $ToolPath)) {
+    $ToolPath = New-TemporaryDirectory
+  }
+  Write-Warning "Installing the ToolCommandName tool under $ToolPath... "
+  dotnet tool install --tool-path $ToolPath --add-source $DevOpsFeed --version $ToolVersion $ToolName | Out-Null
 
-$results = $ownedFolders[$target]
+  return Join-Path $ToolPath $ToolCommandName
+}
 
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function GetCodeOwners ([string] $Command) {
+  # Run code owner tools to retrieve code owners.
+  if (!(Get-Command $Command -errorAction SilentlyContinue)) {
+    Write-Error "The ToolCommandName tool is not properly installed. Please check your tool path. $ToolPath"
+    return ""
+  }
+  
+  # Params $RootDirectory is already in use in cpp release pipeline. 
+  # Will use $CodeOwnerFileLocation and deprecate $RootDirectory once it is ready to retire $RootDirectory.
+  if ($RootDirectory -and !(Test-Path $CodeOwnerFileLocation)) {
+    $CodeOwnerFileLocation = Join-Path $RootDirectory $LocalCodeOwnerPath
+  }
+  
+  $codeOwnersString = & $Command --target-directory "$TargetDirectory" --code-owner-file-path "$CodeOwnerFileLocation" 
+  # Failed at the command of fetching code owners.
+  if ($LASTEXITCODE -ne 0) {
+    return ""
+  }
+  
+  $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
+  if (!$codeOwnersJson) {
+    Write-Host "No code owners returned from the path: $TargetDirectory"
+    return ""
+  }
   
+  $codeOwners = $codeOwnersJson.Owners -join ","
+  Write-Host "Code owners are $codeOwners"
   if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
-
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
-    }
-    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$results"
+    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeOwners"
   }
-
-  return $results
-}
-else {
-  Write-Host "Unable to match path $target in CODEOWNERS file located at $codeOwnersLocation."
-  Write-Host ($ownedFolders | ConvertTo-Json)
-  return ""
 }
+
+# Install the tool first
+$output = InstallRetrieveCodeOwnersTool
+
+return GetCodeOwners -Command $output
```

Comment:
This is just a duplication. We should instead call the InstallRetrieveCodeOwnersTool function directly here. Might want to call it Get-CodeOwnersTool.